flink-connector-http/src/main/java/org/apache/flink/connector/http/table/lookup/HttpTableLookupFunction.java

```diff
@@ -120,11 +120,16 @@ public Collection<RowData> lookup(RowData keyRow) {
             }
         }
         // if we did not get the physical arity from the http response physical row then get it from
-        // the producedDataType, which is set when we have metadata
-        if (physicalArity == -1 && producedDataType != null) {
-            List<LogicalType> childrenLogicalTypes =
-                    producedDataType.getLogicalType().getChildren();
-            physicalArity = childrenLogicalTypes.size() - metadataArity;
+        // the producedDataType. which is set when we have metadata or when there's no data
+        if (physicalArity == -1) {
+            if (producedDataType != null) {
```

Comment:
Lets invert this check, so we can return empty first, and then spare the extra indentation level for the code that updates `physicalArity`.